llvm/lib/CodeGen/RegisterCoalescer.cpp

```diff
@@ -1190,7 +1203,15 @@ bool RegisterCoalescer::removePartialRedundancy(const CoalescerPair &CP,
     // If the newly created Instruction has an address of an instruction that was
     // deleted before (object recycled by the allocator) it needs to be removed from
```

Comment:
I don't really trust this system of relying on deleted instruction pointers. Is there a better way to handle this?